src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1448,6 +1497,21 @@ private RefreshItem getRefreshToken(final AuthenticationRequest request) {
                 item = mTokenCacheStore.getItem(keyUsed);
                 multiResource = true;
             }
+            
+            // If still cannot find an item, try with family clientid
+            // Disable this feature for ADFS authority
+            if (item == null || StringExtensions.IsNullOrBlank(item.getRefreshToken()))
+            {
+                if (!UrlExtensions.isADFSAuthority(StringExtensions.getUrl(mAuthority)))
+                {
+                    Logger.v(TAG + methodName, "No refresh token found, trying to find family client id item.");
+                    item = findFamilyItemForUser(request.getUserIdentifierType(), userId);
+                }
+                else 
+                {
+                    Logger.v(TAG + methodName, "No refresh token found, skip the family client id item lookup for ADFS authority.");
+                }
```

Comment:
you don't actually set keyUsed and your RefreshItem will be created with MultiResourceRefreshTokenKey